File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/dtos/FAQRequest.java
package dev.jefy.connectpro.portfolio.applicaion.dtos;


import org.springframework.util.Assert;

/**
 * @author Jôph Yamba
 */
public record FAQRequest(
    String question,
    String answer
) {
    public FAQRequest {
        Assert.hasText(question, "question must not be empty");
        Assert.hasText(answer, "answer must not be empty");
    }
}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/dtos/FAQRequest.kt
package dev.jefy.connectpro.portfolio.application.dtos

import jakarta.validation.constraints.NotBlank

data class FAQRequest(
    @field:NotBlank(message = "question must not be empty")
    val question: String,
    @field:NotBlank(message = "answer must not be empty")
    val answer: String
){
    init {
        require(question.isNotBlank()) { "question must not be blank" }
        require(answer.isNotBlank()) { "answer must not be blank" }
    }
}